### PR TITLE
bump yabridge: 3.3.1 -> 3.5.2; also add bitbridge support

### LINKED ISSUE #132935: yabridge 32-bit bitbridge support
The current version of [yabridge packaged](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/audio/yabridge/default.nix) doesn't build support for the bitbridge that allows for running 32 bit VST plugins. Information on building that is on the [project README](https://github.com/robbert-vdh/yabridge#32-bit-bitbridge).
Basically, what's necessary is to make sure that the 32 bit versions of `boost` and `libxcb` are installed, and adding a meson flag.
It didn't end up being that simple when I tried adding it myself by just adding the 32 bit dependencies to buildInputs and the meson flag; I got a vague error about meson being unable to find boost_filesystem.

paging the maintainer of the package: @kira-bruneau

### PR DESCRIPTION
**Pull request by critbase**
###### Motivation for this change
Bump yabridge and yabridgectl to the latest version; also add in bitbridge support.

closes #132935
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
@ofborg eval

File: pkgs/tools/audio/yabridge/default.nix [hunk 1]
```diff
@@ -118,11 +120,16 @@ in stdenv.mkDerivation rec {
     "-Dtomlplusplus:BUILD_TESTS=disabled"
   ];
 
+  preConfigure = ''
+    sed -i "214s|xcb.*|xcb_32bit_dep = winegcc.find_library('xcb', dirs: [ '${lib.getLib pkgsi686Linux.xorg.libxcb}/lib', ])|" meson.build
+    sed -i "192 i '${lib.getLib pkgsi686Linux.boost}/lib'," meson.build
+  '';
+
```
Comment (kira-bruneau) [inline pkgs/tools/audio/yabridge/default.nix:127]: Wow, this workaround is simpler than I thought! Thanks for looking into this :heart:

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 133575` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>yabridge</li>
    <li>yabridgectl</li>
  </ul>
</details>

**Review by kira-bruneau — COMMENTED** [review]
@critbase What VST have you been testing with? I've been running into a stack overflow crash with Upright Piano & Reaper since 3.4.0 which is why I've been holding off on updates. I thought it was a more general issue, but maybe it's specific to the plugin?

**Comment by critbase** [comment]
@kira-bruneau I tested with some freeware guitar amp sims (Blue Cat Axiom, LePou Lextac), and I didn't experience a crash or anything. I also downloaded (what i assume is) the same upright piano plugin (https://99sounds.org/vst-plugins/?) and tested it with reaper. worked fine for me.. might be worth testing this PR to make sure though.

**Review by kira-bruneau — APPROVED** [review]
I tried, and it was still crashing. Something probably got messed up on my system. If it's working for you though, then I think this would be good to merge!

**Pull request merged**